tensorflow_datasets/image/ade20k.py

```diff
@@ -0,0 +1,364 @@
+    "television receiver, television, television set, tv, tv set, idiot "
+    "box, boob tube, telly, goggle box", "airplane, aeroplane, plane",
+    "dirt track", "apparel, wearing apparel, dress, clothes", "pole",
+    "land, ground, soil",
+    "bannister, banister, balustrade, balusters, handrail",
+    "escalator, moving staircase, moving stairway",
+    "ottoman, pouf, pouffe, puff, hassock", "bottle",
+    "buffet, counter, sideboard",
+    "poster, posting, placard, notice, bill, card", "stage", "van", "ship",
+    "fountain",
+    "conveyer belt, conveyor belt, conveyer, conveyor, transporter",
+    "canopy", "washer, automatic washer, washing machine", "plaything, toy",
+    "swimming pool, swimming bath, natatorium", "stool", "barrel, cask",
+    "basket, handbasket", "waterfall, falls", "tent, collapsible shelter",
+    "bag", "minibike, motorbike", "cradle", "oven", "ball",
+    "food, solid food", "step, stair", "tank, storage tank",
+    "trade name, brand name, brand, marque", "microwave, microwave oven",
+    "pot, flowerpot",
+    "animal, animate being, beast, brute, creature, fauna",
+    "bicycle, bike, wheel, cycle", "lake",
+    "dishwasher, dish washer, dishwashing machine",
+    "screen, silver screen, projection screen", "blanket, cover",
+    "sculpture", "hood, exhaust hood", "sconce", "vase",
+    "traffic light, traffic signal, stoplight", "tray",
+    "ashcan, trash can, garbage can, wastebin, ash bin, ash-bin, ashbin, "
+    "dustbin, trash barrel, trash bin", "fan",
+    "pier, wharf, wharfage, dock", "crt screen", "plate",
+    "monitor, monitoring device", "bulletin board, notice board", "shower",
+    "radiator", "glass, drinking glass", "clock", "flag")
+
+
+class Ade20k(tfds.core.GeneratorBasedBuilder):
+  """
+  Semantic/part segmentation dataset.
+
+  Segmentation masks are stored as 1-channel png images.
+  """
+
+  VERSION = tfds.core.Version('0.1.0')
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict(dict(
+            image=tfds.features.Image(
+                shape=(None, None, 3), dtype=tf.uint8, encoding_format='jpeg'),
+            segmentation=tfds.features.Image(
+                shape=(None, None, 1), dtype=tf.uint8, encoding_format='png'),
+            example_id=tfds.features.Text(),
+        )),
+        supervised_keys=('image', 'segmentation'),
+        urls=['https://groups.csail.mit.edu/vision/datasets/ADE20K/'],
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    base_dir = dl_manager.download_and_extract(
+        "http://data.csail.mit.edu/places/ADEchallenge/ADEChallengeData2016.zip")
+
+    dirs = '%s/ADEChallengeData2016/{data}/{split}/' % base_dir
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=16,
```

Comment:
`num_shards` is no longer needed, it is computed automatically. Please remove it.

Reply:
Excellent - I never really had a good way of choosing that anyway :)